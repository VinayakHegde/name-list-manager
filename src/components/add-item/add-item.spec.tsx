import * as React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import AddItem from ".";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const withProps = (props) => {
  return <AddItem {...props} />;
};

describe("<AddItem />", () => {
  const props = {
    list: ["a", "b", "c"],
    add: jest.fn(),
  };
  it("should match the snapshot", async () => {
    act(() => {
      render(withProps(props), container);
    });
    expect(container).toMatchSnapshot();
    expect(container.querySelector("button").disabled).toBeTruthy();
  });
});
