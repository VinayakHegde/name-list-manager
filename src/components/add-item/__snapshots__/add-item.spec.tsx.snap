// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddItem /> should match the snapshot 1`] = `
<div>
  <form
    class="addItem"
    data-testid="addItem"
  >
    <input
      placeholder="Enter a name"
      value=""
    />
    <button
      class="primary"
      disabled=""
      type="submit"
    >
      Add
    </button>
    <div
      class="addItem__notice"
    >
      Number of items in the list: 3
    </div>
  </form>
</div>
`;
